compiler/desugared/linting.ml

```diff
@@ -44,6 +44,62 @@ let detect_empty_definitions (p : program) : unit =
         scope.scope_defs)
     p.program_scopes
 
+(* To detect rules that have the same justification and conclusion, we create a
+   set data structure with an appropriate comparison function *)
+module RuleExpressionsMap = Map.Make (struct
+  type t = rule
+
+  let compare x y =
+    let xj, xj_mark = x.rule_just in
+    let yj, yj_mark = y.rule_just in
+    let just =
```

Comment:
Note that in `ast.ml` the unbox is done before the comparison ; the two approaches should be equivalent.
A shorter version would cheat by using `Expr.compare` on `EDefault {exceptions=[]; just; cons}`, but meh ;)

Reply:
Here we're doing only one `unbox` instead of two, which seems more optimal to me because `unbox` appears to be costly.

Reply:
I applied the same "one-unboxing-only" in `ast.ml`.